modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/cards/Cards.tsx

```diff
@@ -56,6 +61,37 @@ const Card = ({item, schema}: {item: any; schema: ICardSchema}) => {
 		actionsRef.current &&
 		(filterItemActions(actionsRef.current, item) as any);
 
+	const getLabels = (
+		item: any
+	): Array<{
+		displayType: TDisplayType;
+		value: string;
+	}> => {
+		if (!schema.labels || !Array.isArray(schema.labels)) {
+			return [];
+		}
+
+		return schema.labels
+			.map((label: ICardLabelSchema) => {
+				const {displayKey, displayMapping} = label;
+				let {displayType} = label;
+
+				if (!displayType && displayMapping && displayKey) {
+					const keyValue = getLocalizedValue(item, displayKey)?.value;
+
+					displayType = displayMapping[keyValue!];
+				}
+
+				const value = getLocalizedValue(item, label.value)?.value || '';
+
+				return {
+					displayType: displayType!,
+					value,
+				};
+			})
+			.filter((label) => label.value);
```

Comment:
Any special reason for this filtering? It also seems a bit overprotective. `value` is set as mandatory in TS, that should be enough,

Reply:
With filtering
<img width="537" alt="image" src="https://github.com/user-attachments/assets/10fbe67d-ef47-4fd9-88fd-799485e357e9" />


Without filtering
<img width="549" alt="image" src="https://github.com/user-attachments/assets/c3561a98-6b45-4563-a3cc-be328312f8c1" />

Reply:
Got it :+1: 

The only change I'd make here then is to do this filtering first, so more complex `.map`ping is only done where is makes sense.

Reply:
Is this is the case where we have incomplete mapping, or if value is not set?

If it's about value not being set, than it's more then fine to filter it out. I would just move that filtering before `.map`, for tiny performance gain.

But if this is about incomplete mapping, I am not sure this is the correct expected behavior. It might be that we want to show it in some default.

Reply:
I'll investigate and report back

Reply:
This is the case where the color value is not set.

Reply:
Because the value is being calculated in the map, we can't put the filter above the map, but I came up with a solution using `flatMap` to have just one loop